backend/api/Services/RobotService.cs

```diff
@@ -62,9 +67,9 @@ public async Task<Robot> CreateFromQuery(CreateRobotQuery robotQuery)
                 };
                 context.Entry(robotModel).State = EntityState.Unchanged;
                 await context.Robots.AddAsync(newRobot);
-                await context.SaveChangesAsync();
-                _ = signalRService.SendMessageAsync("Robot list updated", GetEnabledRobotsWithSubModels().Select((r) => new RobotResponse(r)));
-                return newRobot;
+                await ApplyDatabaseUpdate(newRobot.CurrentInstallation);
+                _ = signalRService.SendMessageAsync("Robot added", newRobot?.CurrentInstallation, new RobotResponse(newRobot!));
```

Comment:
newRobot should either be followed by a "?" or a "!". Would argue that it should not be a question mark.

Reply:
Yes I suppose we should assume that newRobot cannot be null